src/v/kafka/server/handlers/describe_acls.cc

```diff
@@ -24,55 +26,53 @@
 
 namespace kafka {
 
-static void fill_response(
-  request_context& ctx,
-  security::acl_binding_filter& filter,
-  describe_acls_response_data& response) {
-    /*
-     * collapse common acls by pattern
-     */
-    absl::flat_hash_map<
-      security::resource_pattern,
-      std::vector<security::acl_entry>>
-      entries;
-
-    auto bindings = ctx.authorizer().acls(filter);
-
-    for (const auto& binding : bindings) {
-        entries[binding.pattern()].push_back(binding.entry());
-    }
+class describe_acls_exception : public std::exception {
+public:
+    explicit describe_acls_exception(std::error_code ec)
+      : _ec(ec) {}
+
+    cluster::errc error_code() const { return cluster::errc(_ec.value()); }
```

Comment:
This conversion is unsafe, too. It might be simpler to avoid this exception altogether - it's only used in a single place to skip over a single line of code. You're still left with converting a `std::error_code` to a `cluster::errc` somewhere, though.

Reply:
Fixed